crate/operator/operations.py

```diff
@@ -681,6 +696,73 @@ async def suspend_or_start_cluster(
                     )
                 await check_all_data_nodes_gone(core, namespace, name, old)
 
+                # Try to delete the load balancing service if present
+                await delete_lb_service(core, namespace, name)
+
+
+async def _get_connection_factory(core, namespace: str, name: str):
+    """
+    Returns a connection factory.
+    Requires the load balancer to be ready.
+    """
+    host = await get_host(core, namespace, name)
+    password = await get_system_user_password(core, namespace, name)
+    conn_factory = connection_factory(host, password)
+    return conn_factory
+
+
+async def delete_lb_service(core: CoreV1Api, namespace: str, name: str):
+    svc: V1Service = await get_lb_service(core, namespace, name)
+
+    if svc:
+        svc_name = f"crate-{name}"
+        await core.delete_namespaced_service(
+            name=svc_name, namespace=namespace, body=V1DeleteOptions()
+        )
+
+    # If the Load balancer service was already not present or a delete call did not
+    # trigger an exception we consider the operation a success
+    return True
```

Comment:
Might as well not returning anything, if it's always True? Nothing is checking this value either.